pallav/fashionTech/pallavBanka/parkingAssignment/Parking.java

```diff
@@ -0,0 +1,122 @@
+package fashionTech.pallavBanka.parkingAssignment;
+
+import java.util.*;
+
+class ParkingCharge {
+    Date inTime;
+    Date outTime;
+    private Long ParkingTime(Date inTime, Date outTime) {
+        this.inTime = inTime;
+        this.outTime = outTime;
+        if (this.outTime != null) {
+            //difference in time in milliseconds
+            Long diff = outTime.getTime() - inTime.getTime();
+            Long diffHours = diff / (60 * 60 * 1000) % 24;
+            return diffHours;
+        } else{
+            System.out.println("The vehicle hasn't left the premises.");
+            return null;
+        }
+    }
+    private Long charges(String s){
+        Hashtable<String, Integer> charges=new Hashtable<String, Integer>();
+        charges.put("TwoWheeler", 20);
```

Comment:
Try to use constants for these strings, and initialize them in the begning, look for static block in Java
Also, don't use HashTable, use HashMap
Map<String, Integer> charges = new HashMap();

Reply:
In this method you should be only fetching data from this hashMap and calculating, what is the point of putting values in HashMap in this method and then fetching again.

Reply:
But first the hashmap should have something to fetch from